packages/devtools_app/test/network/offline_data_test.dart

```diff
@@ -0,0 +1,181 @@
+      expect((serializedJson['socket'] as Map)['writeBytes'], 18237);
+    });
+
+    test('Socket duration should be calculated correctly', () {
+      final expectedDuration = Duration(
+        microseconds: firstSocket.endTimestamp!.microsecondsSinceEpoch -
+            firstSocket.startTimestamp.microsecondsSinceEpoch,
+      );
+
+      expect(firstSocket.duration, expectedDuration);
+    });
+
+    test('Socket status should indicate "Open" or "Closed" based on endTime',
+        () {
+      expect(
+        firstSocket.status,
+        'Closed',
+      ); // The provided socket has an endTime
+
+      // Modify socket to simulate "Open" status
+      final openSocketJson = {
+        ...firstSocket.toJson(),
+        'socket': {
+          ...(firstSocket.toJson()['socket'] as Map<String, Object?>),
+          'endTime': null,
+        },
+      };
+      final openSocket = Socket.fromJson(openSocketJson);
+
+      expect(openSocket.status, 'Open'); // No endTime indicates "Open"
+    });
+
+    test('Socket equality and hash code should work correctly', () {
+      expect(firstSocket == secondSocket, isFalse);
+      expect(firstSocket.hashCode != secondSocket.hashCode, isTrue);
+
+      final duplicateSocket = Socket.fromJson(firstSocket.toJson());
+      expect(firstSocket, equals(duplicateSocket));
+      expect(firstSocket.hashCode, equals(duplicateSocket.hashCode));
+    });
+  });
+
+  group('OfflineNetworkData Tests', () {
+    test('OfflineNetworkData should deserialize correctly', () {
+      // Validate httpRequestData
+      expect(offlineData.httpRequestData.length, 2);
+      expect(offlineData.httpRequestData.first.id, '975585676925010898');
+      expect(offlineData.httpRequestData.first.method, 'GET');
+      expect(
+        offlineData.httpRequestData.first.uri,
+        'https://jsonplaceholder.typicode.com/albums/1',
+      );
+
+      // Validate socketData
+      expect(offlineData.socketData.length, 2);
+      expect(offlineData.socketData.first.id, '105553123901536');
+      expect(offlineData.socketData.first.socketType, 'tcp');
+      expect(offlineData.socketData.first.port, 443);
+
+      // Validate selectedRequestId
+      expect(offlineData.selectedRequestId, isNull);
+    });
+
+    test('OfflineNetworkData should serialize correctly', () {
+      final serializedJson = offlineData.toJson();
+
+      // Validate serialized JSON
+      final httpRequestData = serializedJson['httpRequestData'] as List;
+      final firstRequest = httpRequestData.first as Map<String, Object?>;
+      final requestDetails = firstRequest['request'] as Map<String, Object?>;
+
+      expect(requestDetails['id'], '975585676925010898');
+
+      final socketData = serializedJson['socketData'] as List;
+      final firstSocket = socketData.first as Map<String, Object?>;
+      final socketDetails = firstSocket['socket'] as Map<String, Object?>;
+
+      expect(socketDetails['id'], '105553123901536');
```

Comment:
same here: we already verified the socket data above, so these can be removed

Reply:
done